src/java/picard/analysis/artifacts/ContextAccumulator.java

```diff
@@ -48,9 +42,12 @@ public void countRecord(final String refContext, final char calledBase, final SA
      */
     public ListMap<Transition, DetailPair> calculateMetrics(final String sampleAlias, final String library) {
         final ListMap<Transition, DetailPair> detailMetricsMap = new ListMap<>();
-        for (final Transition altTransition : Transition.altValues()) {
-            final Transition refTransition = altTransition.matchingRef();
-            for (final String context : new TreeSet<>(this.artifactMap.get(altTransition).keySet())) {
+        for (final String context : new TreeSet<>(artifactMap.keySet())) {
+
+            final char refBase = context.charAt(context.length() / 2);
```

Comment:
Actually, the modulo operator is pretty time consuming.  However, it was bad when it was being calculated many times per base per read; but now that it's only at the end it won't be that bad.  That being said, do you really want to put the check here and fail out after processing the entire genome?  Why not put the check when the artifactMap is created?  In fact, that's what I'll do.